mm/2s2h/Enhancements/Minigames/Archery.cpp

```diff
@@ -51,4 +59,31 @@ void RegisterArchery() {
     });
 }
 
-static RegisterShipInitFunc initFunc(RegisterArchery, { SWAMP_CVAR_NAME, TOWN_CVAR_NAME });
+static void ResetBoatArcheryScore(UNUSED Actor* actor) {
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_25_20)) {
+        HS_SET_BOAT_ARCHERY_HIGH_SCORE(BOAT_CVAR - 1);
+    }
+}
+
+static void RegisterBoatArchery() {
+    COND_ID_HOOK(OnActorInit, ACTOR_EN_DNH, BOAT_CVAR != 20, ResetBoatArcheryScore);
+}
+
+static bool ShouldFailBoatArchery() {
+    return gSaveContext.minigameHiddenScore >= BOAT_HEALTH_CVAR;
+}
+
+static void RegisterKoumeHealth() {
+    REGISTER_VB_SHOULD(VB_FAIL_BOAT_ARCHERY, { *should = ShouldFailBoatArchery(); });
+}
+
+static void RegisterKoumeInvincible() {
+    COND_VB_SHOULD(VB_KOUME_TAKE_DAMAGE, BOAT_NO_DAMAGE_CVAR, { *should = false; });
+}
+
+static RegisterShipInitFunc initFunc_Swamp(RegisterSwampArchery, { SWAMP_CVAR_NAME });
+static RegisterShipInitFunc initFunc_Town(RegisterTownArchery, { TOWN_CVAR_NAME });
+
+static RegisterShipInitFunc initFunc_Boat(RegisterBoatArchery, { BOAT_CVAR_NAME });
+static RegisterShipInitFunc initFunc_Health(RegisterKoumeHealth);
```

Comment:
And then make the update path for this `BOAT_HEALTH_CVAR_NAME`. That way Koume's health gets updated immediately.
```suggestion
static RegisterShipInitFunc initFunc_Health(RegisterKoumeHealth, { BOAT_HEALTH_CVAR_NAME });
```